insertion_sort.rb

```diff
@@ -0,0 +1,16 @@
+# https://en.wikipedia.org/wiki/Insertion_sort
+def insertion_sort a
+  for i in 1..(a.size - 1)
+    x = a[i]
+    j = i
+    while(j > 0 && a[j-1] >= x)
+      a[j] = a[j-1]
```

Comment:
Holding a temporary variable makes it so you only need to swap once as opposed to every iteration in the while loop which I think makes it (negligibly) faster than using ruby's swap 

Here is an example
<pre>
[1,2,3,1]
</pre>

Two swaps
<pre>
[1,2,3,1]
[1,2,1,3]
[1,1,2,3]
</pre>

One swap
<pre>
[1,2,1,3]
[1,2,2,3]
[1,1,2,3]
</pre>